contracts/neutron_interchain_queries/src/contract.rs

```diff
@@ -238,10 +252,33 @@ pub fn sudo_tx_query_result(
                     for coin in transfer_msg.amount {
                         deposits.push(Transfer {
                             sender: transfer_msg.from_address.clone(),
-                            amount: coin.amount,
+                            amount: coin.amount.clone(),
                             denom: coin.denom,
                             recipient: recipient.to_string(),
                         });
+                        let mut stored_transfers: u64 =
+                            TRANSFERS.load(deps.storage).unwrap_or_default();
+                        stored_transfers += 1;
+                        TRANSFERS.save(deps.storage, &stored_transfers)?;
```

Comment:
1. should we move `TRANSFERS.save` into `Ok` arm? 
2. do you think we can a bit refactor this fn into several fns so it makes a bit more sense?

Reply:
you mean the fn is too long? If so, I wonder if we need these debug logs, they take from a quarter to a half of the fn. We now have queries and mutable contract's state, and it can be used to track sudo execution. Let me delete debug logs and split up fn a bit.